src/client/src/components/AddTaskForm/AddTaskForm.jsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import ItemClient from "../../services/taskService";
+
+import "./addTaskForm.css";
+
+const AddTaskForm = ({
+  tasks,
+  setTasks,
+  editTask,
+  setEditTask,
+  setErrorMessage,
+  setIsErrorShown,
+}) => {
+  const [input, setInput] = useState("");
+
+  const taskService = new ItemClient();
+  let iconClassName;
+
+  useEffect(() => {
+    if (editTask) {
+      setInput(editTask.itemName);
+    } else {
+      setInput("");
+    }
+  }, [editTask, setInput]);
+
+  const setButtonIcon = () => {
+    if (editTask) {
+      iconClassName = "fa fa-pencil-alt";
+    } else {
+      iconClassName = "fa fa-plus";
+    }
+  };
+
+  const updateTask = async (input, id, itemName, status) => {
+    const taskToEdit = tasks.find((task) => task.id === id);
+    taskToEdit.itemName = input;
+    taskToEdit.status = status;
+    const isEdited = await taskService.updateTask(taskToEdit);
+    if (isEdited) {
+      setTasks(
+        tasks.map((task) => {
+          if (task.id === id) {
+            return taskToEdit;
```

Comment:
a better thing is to use the response from the server and not the data from the request.
The server can actually return different stuff then what was requested, for example when sending "1" and actually adding a pokemon task.